ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -182,46 +207,30 @@ impl AotIRDisplay for BinOp {
     }
 }
 
-/// An instruction opcode.
-#[deku_derive(DekuRead)]
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-#[deku(type = "u8")]
-pub(crate) enum Opcode {
-    Nop = 0,
-    Load,
-    Store,
-    Alloca,
-    Call,
-    Br,
-    CondBr,
-    Icmp,
-    Ret,
-    InsertValue,
-    PtrAdd,
-    BinOp,
-    Unimplemented = 255,
-}
-
-impl AotIRDisplay for Opcode {
-    fn to_string(&self, _m: &Module) -> String {
-        format!("{:?}", self).to_lowercase()
-    }
-}
-
 /// Uniquely identifies an instruction within a [Module].
 #[deku_derive(DekuRead)]
 #[derive(Clone, Debug, Hash, Eq, PartialEq)]
 pub(crate) struct InstructionID {
-    #[deku(skip)] // computed after deserialisation.
-    func_idx: FuncIdx,
+    /// The index of the parent function.
+    ///
+    /// This is computed after deserialisation and relies on
+    /// `Module::compute_local_operand_func_indices() to correctly identify and update all local
+    /// variable operands, setting this field to `Some`. By using `Option` we can detect the case
+    /// where an operand is missed. Without `Option`, deku would insert a default 0 FuncIdx,
+    /// leading to general chaos.
+    ///
+    /// FIXME: I think we can avoid computing this after the fact by having ykllvm include the func
+    /// index in the serialised IR for all local variable operands.
+    #[deku(skip)]
+    func_idx: Option<FuncIdx>,
```

Comment:
Agreed: this doesn't feel quite right, but it's probably OK for now.

Reply:
I will fix this one in a follow up PR. I have a plan.